core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val stateLock = new ReentrantLock
 
   val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
+  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
```

Comment:
I just looked up what the semantics of "cores.max" are and they seem to be slightly different than what this patch includes:
http://spark.apache.org/docs/latest/running-on-mesos.html#coarse-grained

It says that if "cores.max" is not included, then spark will just blindly accept all cpu offers it is given. In the current patch, *no* gpu resources will be accepted if "gpus.max" is not set.